### DISCUSSION TITLE
Does the Api-Portal allow for default clientId or scopes?

### DISCUSSION THREAD

**Discussion by dallas-fletchall** [mia-platform/community#538] [Q&A] [answered]
### What product are you having troubles with?

Console

### What Console version are you using?

1.102.8

### Description

When using `authcode+pkce` authorization scheme as described [here](https://github.com/mia-platform/community/discussions/518), I am unable to set the default `client_id` or `scopes`.

![image](https://github.com/user-attachments/assets/0bdc16fc-feef-4d6d-a738-52f6a579b48b)

In `swagger-ui` (`docker.io/swaggerapi/swagger-ui:latest`), this can be accomplished by setting the `OAUTH_CLIENT_ID` and `OAUTH_SCOPES` env vars, see [docs](https://swagger.io/docs/open-source-tools/swagger-ui/usage/oauth2/).

Currently I am running `2.1.0-rc4` of the Api-Portal and `3.6.0` of the Swagger-Aggregator. My advanced configuration is as follows:
```json
{
  "baseSwagger": {
        "swagger": "2.0",
        "consumes": [
            "application/json"
        ],
        "produces": [
            "application/json"
        ],
        "security": [
            {
                "oidc": []
            }
        ],
        "securityDefinitions": {
            "oidc": {
                "type": "oauth2",
                "flow": "accessCode",
                "authorizationUrl": "https://example.com/authorize",
                "tokenUrl": "https://example.com/token",
                "scopes": {
                    "openid": "openId scope",
                    "profile": "profile scope"
                }
            }
        }
    }
}
```

I am hoping to auto-complete these to some default so that my users do not have to know the clientId and scopes to use the Documentations page.

### Actual Outcome

Users have to know the associated client, and go to the IdP to retrieve the clientId.

### Expected Outcome

A default value can be configured, and is auto-populated. The user can decide to overwrite if necessary.

**Comment by epessina** [comment, marked as answer]
Hi @dallas-fletchall , thanks for reaching out.

With the newly released API Portal version 2.1.0 and Swagger Aggregator version 3.8.0, you can [provide](https://next.docs.mia-platform.eu/docs/runtime_suite_applications/api-documentation-aggregator/faqs#how-can-i-configure-my-oauth-20-flow) a swagger-ui OAuth 2.0 configuration to achieve your desired behaviour.

In your case, the advanced configuration should look something like:

```json
{
  "baseSwagger": {
        "swagger": "2.0",
        "consumes": [
            "application/json"
        ],
        "produces": [
            "application/json"
        ],
        "security": [
            {
                "oidc": []
            }
        ],
        "securityDefinitions": {
            "oidc": {
                "type": "oauth2",
                "flow": "accessCode",
                "authorizationUrl": "https://example.com/authorize",
                "tokenUrl": "https://example.com/token",
                "scopes": {
                    "openid": "openId scope",
                    "profile": "profile scope"
                }
            }
        }
    },
    "oauthConfig": {
      "usePkceWithAuthorizationCodeGrant": true,
      "clientId": "your-default-client-id",
      "scopes": ["your-default-scope"]
    }
}
```

Please, let me know if you can achieve the expected behaviour with this new feature

**Reply by dallas-fletchall** [reply]
I was able to verify that the default scopes and default clientId worked on the new versions of the plugins (api-portal `2.1.0` & swagger-aggregator `3.8.0`).

Thanks!
